api/v1alpha1/lvmcluster_webhook.go

```diff
@@ -277,18 +326,21 @@ func (l *LVMCluster) verifyNoDeviceOverlap() error {
 	return nil
 }
 
-func (l *LVMCluster) getPathsOfDeviceClass(deviceClassName string) ([]string, error) {
-
+func (l *LVMCluster) getPathsOfDeviceClass(deviceClassName string) (required []string, optional []string, err error) {
+	required, optional, err = []string{}, []string{}, nil
 	for _, deviceClass := range l.Spec.Storage.DeviceClasses {
 		if deviceClass.Name == deviceClassName {
 			if deviceClass.DeviceSelector != nil {
-				return deviceClass.DeviceSelector.Paths, nil
+				required = deviceClass.DeviceSelector.Paths
```

Comment:
@qJkee I think they are separate here because we want to return if we find the right device class regardless if there is a device selector. We only want to set `required` and `optional if there is a device selector. So no matter what I think we end up with 2 if statements. I at least can't think of a way to combine these but retain device selector not being required